internal/command/describe_type.go

```diff
@@ -0,0 +1,145 @@
+ * limitations under the License.
+ */
+
+package command
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/apache/camel-k/pkg/apis/camel/v1alpha1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"knative.dev/client/pkg/printers"
+	"knative.dev/pkg/apis"
+
+	knerrors "knative.dev/client/pkg/errors"
+	"knative.dev/client/pkg/kn/commands"
+
+	"github.com/spf13/cobra"
+)
+
+var describeExample = `
+  # Describe given Kamelets
+  kn-source-kamelet describe-type NAME
+
+  # Describe given Kamelets in YAML output format
+  kn-source-kamelet describe-type NAME -o yaml`
+
+// NewDescribeTypeCommand implements 'kn-source-kamelet describe-type' command
+func NewDescribeTypeCommand(p *KameletPluginParams) *cobra.Command {
+	printFlags := genericclioptions.NewPrintFlags("")
+
+	cmd := &cobra.Command{
+		Use:     "describe-type",
+		Short:   "Show details of given Kamelet source type",
+		Aliases: []string{"dt"},
+		Example: describeExample,
+		RunE: func(cmd *cobra.Command, args []string) (err error) {
+			if len(args) != 1 {
+				return errors.New("'kn-source-kamelet describe-type' requires the Kamelet name given as single argument")
+			}
+			name := args[0]
+
+			namespace, err := p.GetNamespace(cmd)
+			if err != nil {
+				return err
+			}
+
+			client, err := p.NewKameletClient()
+			if err != nil {
+				return err
+			}
+
+			kamelet, err := client.Kamelets(namespace).Get(p.Context, name, v1.GetOptions{})
+			if err != nil {
+				return knerrors.GetError(err)
+			}
+
+			out := cmd.OutOrStdout()
+
+			if printFlags.OutputFlagSpecified() {
+				if strings.ToLower(*printFlags.OutputFormat) == "url" {
+					fmt.Fprintf(out, "%s\n", kamelet.GetSelfLink())
+					return nil
+				}
+				printer, err := printFlags.ToPrinter()
+				if err != nil {
+					return err
+				}
+				return printer.PrintObj(kamelet, out)
+			}
+
+			dw := printers.NewPrefixWriter(out)
+
+			printDetails, err := cmd.Flags().GetBool("verbose")
+			if err != nil {
+				return err
+			}
+
+			writeKamelet(dw, kamelet, printDetails)
+			dw.WriteLine()
+			if err := dw.Flush(); err != nil {
+				return err
+			}
+
+			// Condition info
+			commands.WriteConditions(dw, asApiConditions(kamelet.Status.Conditions), printDetails)
+			if err := dw.Flush(); err != nil {
+				return err
+			}
+
+			return nil
+		},
+	}
+	flags := cmd.Flags()
+	commands.AddNamespaceFlags(flags, false)
+	flags.BoolP("verbose", "v", false, "More output.")
+	printFlags.AddFlags(cmd)
+	cmd.Flag("output").Usage = fmt.Sprintf("Output format. One of: %s.", strings.Join(append(printFlags.AllowedFormats(), "url"), "|"))
+	return cmd
+}
+
+func writeKamelet(dw printers.PrefixWriter, kamelet *v1alpha1.Kamelet, printDetails bool) {
+	commands.WriteMetadata(dw, &kamelet.ObjectMeta, printDetails)
+	dw.WriteAttribute("Type", extractType(kamelet))
+	dw.WriteAttribute("Phase", string(kamelet.Status.Phase))
+	dw.WriteAttribute("Title", kamelet.Spec.Definition.Title)
```

Comment:
What is 'Title' ? Is it a short description of the Kamelet ? Maybe this could be defined within the "Description" section, so that is used as kind of the title of the description :)

Reply:
I know it might be difficult but I would try to move the description updwards and close to the name, as I think they belong together and are more important then other details like labels and annotations.